src/dataServices/dynamoDbUtil.ts

```diff
@@ -39,6 +42,19 @@ export class DynamoDbUtil {
         }
         item[DOCUMENT_STATUS_FIELD] = documentStatus;
         item[LOCK_END_TS_FIELD] = Date.now();
+
+        // Format of flattenedResource
+        // https://www.npmjs.com/package/flat
+        // flatten({ key1: { keyA: 'valueI' } })  => { key1.keyA: 'valueI'}
+        const flattenedResources: Record<string, string> = flatten(resource);
+        const references = Object.keys(flattenedResources)
+            .filter((key: string) => {
+                return key.endsWith('reference');

```

Comment:
Without the dot it may introduce a bug. Not sure if there's an actual FHIR resource that has a non-reference field that ends with "reference" though.
```suggestion
                return key.endsWith('.reference');

```